ddmrp/models/stock_buffer.py

```diff
@@ -1256,6 +1284,36 @@ def do_auto_procure(self):
             wizard.make_procurement()
         return True
 
+    def action_view_supply_outside_dlt_window(self):
+        cut_date = self._get_incoming_supply_date_limit()
+        if self.item_type == "purchased":
+            pols = self.purchase_line_ids.filtered(
+                lambda l: l.date_planned > fields.Datetime.to_datetime(cut_date)
+            )
```

Comment:
I would look at moves outside dlt and take related purchase_line_id because the date_expected of the move could differ from the pol planned date and it is the move date that is considered.

Moreover, in our case, self.purchase_line_ids cannot be used after PO confirmation as we have 1 buffer monitoring a stock location shared by multiple warehouse. This allows us to monitor the global stock. The receipt picking type on the PO can be changed depending on what warehouse you want to send goods to. My concern is here: https://github.com/OCA/ddmrp/blob/13.0/ddmrp/models/purchase_order.py#L68 with the warehouse on the domain. Maybe the link should be created only when the po is created from the buffer as you did for orderpoint_ids? And not recompute each time a write is called on the po line.

Side remark, the m2m link between POL and buffer as stored link will grow each time a PO is created and each time the execution priority is recomputed, it will browse more and more records. Maybe this here https://github.com/OCA/ddmrp/blob/13.0/ddmrp/models/purchase_order.py#L49 needs also to be reviewed a little bit for performance.

Last question. When the buffer is linked to the po line, why the priority computation is launched before the creation of the link? See here: https://github.com/OCA/ddmrp/blob/13.0/ddmrp/models/purchase_order.py#L29

Reply:
1. Sure, I adapted it to use the move date whenever possible (confirmed PO), otherwise we need to use the date planned in PO lines (RFQ's).
2. This has been a [feature added recently](https://github.com/OCA/ddmrp/pull/71). I tried to cover the most common scenario, but as I knew a perfect fit for everybody was impossible, I include that method as a hook to allow for specific use case customization.
3.  Note that `self`can already be a reduced recordset of PO lines and that that recordset is filtered. As the buffer changes its execution level, you need to update all not closed PO's with the latest value. You should not have too many simultaneous opened purchase orders for a given stock buffer, however I agree that "not closed PO's" sometimes is difficult to define, and the orders remain in `purchase` state even when they are actually shipped and invoiced. We have implemented in some cases a cron job that closes (set status to `done`) PO's already invoiced and delivered.
4. Good catch, I include a commit fixing it in this PR.